skbio/alignment/_tabular_msa.py

```diff
@@ -717,6 +707,92 @@ def reindex(self, key=None, keys=None):
         self._keys = keys_
 
     @experimental(as_of='0.4.0-dev')
+    def append(self, sequence, minter=None):
+        """Append a sequence to the MSA.
+
+        Parameters
+        ----------
+        sequence : alphabet-aware scikit-bio sequence object
+            Sequence to be appended. Must match the dtype of the MSA and the
+            length of the second dimension of the MSA.
+        minter : callable or metadata key, optional
+            If None, the MSA is checked to see if a previous minter has been
+            cached. If not, an exception is raised. If a cached key is found
+            it is used to reindex the MSA.
+
+        Raises
+        ------
+        TypeError
+            If the sequence object is a type that doesn't have an alphabet
+        TypeError
+            If the type of the sequence does not match the dtype of the MSA.
+        ValueError
+            If the length of the sequence does not match the number of
+            positions in the MSA.
+        OperationError
+            If no key is provided, but the MSA has keys, and no cached key
+            is found.
+
+        See Also
+        --------
+        reindex
+
+        Notes
+        -----
+        The MSA is not automatically re-aligned when a sequence is appended.
+        Therefore, this operation is not necessarily meaningful on its own.
+
+        Examples
+        --------
+        >>> from skbio import DNA, TabularMSA
+        >>> msa = TabularMSA([DNA('')])
+        >>> msa.append(DNA(''))
+        >>> msa == TabularMSA([DNA(''), DNA('')])
+        True
+
+        >>> msa = TabularMSA([DNA('', metadata={'id': 'a'})], minter='id')
+        >>> msa.append(DNA('', metadata={'id': 'b'}), minter='id')
+        >>> msa == TabularMSA([DNA('', metadata={'id': 'a'}),
+        ...                    DNA('', metadata={'id': 'b'})], minter='id')
+        True
+        """
+        if minter is None:
+            if self.has_keys():
+                minter = self.minter
+        else:
+            if not self.has_keys():
+                raise OperationError(
+                    "key was provided but MSA does not have keys.")
+
+        self._add_sequence(sequence)
+        self.reindex(minter=minter)
+
+    def _add_sequence(self, sequence):
+        msa_is_empty = not len(self)
+        if msa_is_empty:
+            dtype = type(sequence)
+            if not issubclass(dtype, IUPACSequence):
+                raise TypeError(
+                    "`sequence` must be a scikit-bio sequence object "
+                    "that has an alphabet, not type %r" % dtype.__name__)
+            self._dtype = dtype
+            self._shape = _Shape(sequence=1, position=len(sequence))
+            self._seqs = [sequence]
+        elif type(sequence) is not self.dtype:
```

Comment:
Modified my previous comment. I changed the original code more than I had remembered, so there's a good chance it's not as optimal as before. Probably would be a good idea to get a second opinion.